### ISSUE TITLE
The "studio" does no preserve subproblems order

### ISSUE THREAD

**Issue by layus** [UCL-INGI/INGInious#624]
I recently stumbled on an issue where editing a task would reorder its subtasks.
It is very easu to observe and debug (with the right permissions) in https://inginious.info.ucl.ac.be/admin/LINGI2172/edit/task/Mission_1_SQL_2020.

In that task, the subproblems range from 01 to 16 and are currently displayed in the right order on the task page. But in the editor problems 10-16 are displayed before problems 01-09. The annoying part is that saving the task in that state reorders the subproblems.

As usual, it boild down to having the subproblems stored in a normal dict instead of an ordered dict. Or possibly a list in this case, as it gets converted to json.

As far as I understand the issue, it comes from encoding the problems as a json dictionary for use by javascript.
In the so called "studio" we have
https://github.com/UCL-INGI/INGInious/blob/77dd58972f8395aac902f22b82ccab50c578ab0b/inginious/frontend/templates/course_admin/task_edit.html#L165
which (for some strange reason, possibly called "OrderedDict") gets written in the right order in the html page.
```
<script type="text/javascript">
    delete_subproblem_message = 'Are you sure that you want to delete this subproblem?';
    problem_data = {"01": {"type": "code", "header": "List the names of all the countries.", "name": "Mission 1 - Queries in SQL", "default": "", "language": "SQL"}, "02": ...
```

But developper tools reaveal the issue. When that dict is parsed as a javascript dict, the keys get reordered, which is perfectly valid because javascript does not guarantee the keys order.

![image](https://user-images.githubusercontent.com/632767/107225497-26a4e700-6a19-11eb-8475-cb0d48252efc.png)

This makes the foreach loop in studio_load incorrect 

https://github.com/UCL-INGI/INGInious/blob/2fc8f0027ec3f6b2e618194eb27896873a610144/inginious/frontend/static/js/studio.js#L7-L17

Oh, here are my tests :-)

![image](https://user-images.githubusercontent.com/632767/107226102-f9a50400-6a19-11eb-9466-f7dd91fec97f.png)

**Comment by anthonygego** [comment]
Fixed in 46830e0857a85ea91e544be5a95926386a19aa77

**Comment by layus** [comment]
Thanks !
